packages/go/dawgs/traversal/traversal.go

```diff
@@ -49,14 +49,18 @@ type PatternMatchDelegate = func(terminal *graph.PathSegment) error
 // The return value of the Do(...) function may be passed directly to a Traversal via a Plan as the Plan.Driver field.
 type PatternContinuation interface {
 	Outbound(criteria ...graph.Criteria) PatternContinuation
+	OutboundWithDepth(min, max int, criteria ...graph.Criteria) PatternContinuation
 	Inbound(criteria ...graph.Criteria) PatternContinuation
+	InboundWithDepth(min, max int, criteria ...graph.Criteria) PatternContinuation
 	Do(delegate PatternMatchDelegate) Driver
 }
 
 // expansion is an internal representation of a path expansion step.
 type expansion struct {
 	criteria  []graph.Criteria
 	direction graph.Direction
+	minDepth  int
+	maxDepth  int
```

Comment:
This might be overly cautious but defining these depths as unsigned integers types might prevent someone from using this incorrectly unless we are handling depths of less than zero.

Reply:
We're not handing it currently, but I think a more correct solution here is to simply set the min/max to the default if its < 0. I'll make the appropriate change